identity-tools/developer_identity/create-dev-identity.sh

```diff
@@ -0,0 +1,256 @@
+                    LwM2M_URL="coaps://mds-systemtest.dev.mbed.com"
+                    API_URL="https://api-os2.mbedcloudstaging.net"
+                    GW_URL="https://gateways.mbedcloudstaging.net"
+                    k8s_URL="https://edge-k8s.mbedcloudstaging.net"
+                    ;;
+                mbedcloud|us-east-1.mbedcloud)
+                    LwM2M_URL="coaps://lwm2m.us-east-1.mbedcloud.com"
+                    API_URL="https://api.us-east-1.mbedcloud.com"
+                    GW_URL="https://gateways.us-east-1.mbedcloud.com"
+                    k8s_URL="https://edge-k8s.us-east-1.mbedcloud.com"
+                    ;;
+                ap-northeast-1.mbedcloud)
+                    LwM2M_URL="coaps://lwm2m.ap-northeast-1.mbedcloud.com"
+                    API_URL="https://api.ap-northeast-1.mbedcloud.com"
+                    GW_URL="https://gateways.ap-northeast-1.mbedcloud.com"
+                    k8s_URL="https://edge-k8s.ap-northeast-1.mbedcloud.com"
+                    ;;
+                *)
+                    cli_error "Unknown mbed-cloud lab instance - $CLOUD_LAB. Check help for expected values."
+                    exit 1
+                    ;;
+            esac
+            ;;
+        c)
+            LwM2M_URL="$OPTARG"
+            ;;
+        g)
+            GW_URL="$OPTARG"
+            ;;
+        s)
+            API_URL="$OPTARG"
+            ;;
+        k)
+            k8s_URL="$OPTARG"
+            ;;
+        p)
+            SERIAL_NUMBER_PREFIX="$OPTARG"
+            ;;
+        n)
+            ACCOUNT_ID="$OPTARG"
+            ;;
+        o)
+            OUTPUT_DIR="$OPTARG"
+            ;;
+        i)
+            DEVICE_ID="$OPTARG"
+            ;;
+        w)
+            HW_VERSION="$OPTARG"
+            ;;
+        r)
+            RADIO_CONFIG="$OPTARG"
+            ;;
+        l)
+            LED_CONFIG="$OPTARG"
+            ;;
+        *)
+            cli_help
+            exit 1
+            ;;
+    esac
+done
+
+shift "$(($OPTIND-1))"
+
+generate_random_hex_number() {
+    hexchars="0123456789ABCDEF"
+    local hex_rand=$( for ((i = 1; i <= $1; i++)) ; do echo -n ${hexchars:$(( $RANDOM % 16 )):1} ; done | sed -e 's/\(..\)/\1/g' )
+    echo "$hex_rand"
+}
+
+SN_POSTFIX="$(generate_random_hex_number 6)"
+
+if [ -z "$ACCOUNT_ID" ]; then
+    cli_warn "-n <account_id> not specified! generating random uuid..."
+    ACCOUNT_ID="$(generate_random_hex_number 32)"
+fi
+
+if [ -z "$DEVICE_ID" ]; then
+    cli_warn "-i <device_id> not specified! generating random uuid..."
+    DEVICE_ID="$(generate_random_hex_number 32)"
+fi
+
+cli_debug "Creating developer self-signed certificate."
+tmpdir=$(mktemp -d)
+. "$DEVID_CLI_DIR/generate_self_signed_certs.sh" $tmpdir
+
+if [ $? -ne 0 ]; then
+    cli_error "Failed to generate self signed certs!"
+    exit 1
+fi
+
+cli_debug "Successfully generated self-signed certs at $tmpdir"
+
+MAC_INDEX_3="$((1 + RANDOM % 250))"
```

Comment:
What's the reason that the MAC address is always randomized? Is it set in the hardware after this script is run?

Reply:
Yes, maestro sets the MAC address of the primary interface. This is legacy code from WigWag where we had to use the WigWag's OUI in the MAC address as we had gone through FCC certification. I think moving on we should not set the MAC address as we are now truly a user space program.